stock_picking_volume/models/stock_picking.py

```diff
@@ -16,12 +16,18 @@ class StockPicking(models.Model):
         string="Volume unit of measure label", compute="_compute_volume_uom_name"
     )
 
-    @api.depends("move_lines", "move_lines.volume")
+    @api.depends("move_lines", "move_lines.volume", "move_lines.state")
     def _compute_volume(self):
         for picking in self:
-            new_volume = sum(picking.move_lines.mapped("volume"))
-            if picking.volume != new_volume:
-                picking.volume = new_volume
+            moves = picking.move_lines
+            exclude_cancel = any(m.state != "cancel" for m in moves)
+            volume = 0
+            for move in moves:
+                if move.state == "cancel" and exclude_cancel:
+                    continue
+                volume += move.volume
```

Comment:
```suggestion
            volume = 0
            for move in moves:
                if move.state == "cancel":
                    continue
                volume += move.volume
```
May be I miss something but IMO this code could be simplified to avoid a useless loop

Reply:
@lmignon no, that's not the same. Purpose is to exclude canceled moves from non-canceled pickings. For canceled pickings, we still compute the volume

Reply:
At least we should have a comment to explain this behavior (And a test for both cases).  If the exclude of cancelled moves from the computed volume makes sense,  I don't see the motivation to still compute the volume of cancelled pickings.  I'm not at ease with the lack of symmetry between the 2 ways of computing the volume.

Reply:
> I don't see the motivation to still compute the volume of cancelled pickings.

To not break current behavior. And I find it good to keep totals on a canceled document.

Reply:
I will not block but this a comment and a test is more than welcome to avoid regression and makes this behavior explicit.

Reply:
I already added a Test for it.

Reply:
there are no tests for the case where we cancel the picking and want to make sure that in this case, the volume is computed on all the moves.